### DISCUSSION TITLE
Edit dotfiles with gedit / admin:// is broken [independly of firejail]

### DISCUSSION THREAD

**Discussion by Froggy232** [netblue30/firejail#4133] [Q&A] [answered]
Hi,

I use firejail system-wide, it work very great but I have one problem : I can't edit the majority of config files with gedit now.
I can open them if they are in my home directory, but I can't write them because it tell me it doesn't have the right to do so.
Also, the `admin://` option to edit system config file doesn't work either with default profile.
Everythings works great with `--noprofile` option.

Do you know what to add to gedit.local to correct that?

Thanks you
Best regards

**Comment by rusty-snake** [comment, marked as answer]
`admin://` is a GIO protocol implemented vis GVFS through D-Bus which uses PolKit for that. (Why am I say this? IDK).
Try `ignore nonewprivs` and/or `ignore noroot`.

EDIT: `firejail gedit admin:///etc/firejail/firejail.config` works for me (firejail from git and Fedora 33). Do you have a gedit.local or globals.local? Or a older firejail version?

> I can open them if they are in my home directory, but I can't write them because it tell me it doesn't have the right to do so.

Either add `ignore include disable-common.inc` or `read-write ${HOME}/.bashrc`+`ignore read-only ${HOME}/.bashrc` (example file).

**Reply by Froggy232** [reply]
`ignore include disable-common.inc` work for the config file in /home, thanks you again!
Do you know if it is possible to apply `read-write ${HOME}/.bashrc` + `ignore read-only ${HOME}/.bashrc` for all files?

I try again, and gedit `admin://` doesn't work with `--noprofile` either, so the problem is not firejail.
I will investigate elsewhere, thanks you a lot!

**Reply by rusty-snake** [reply]
> Do you know if it is possible to apply read-write ${HOME}/.bashrc + ignore read-only ${HOME}/.bashrc for all files?

Assuming you mean in a simpler way then writing it umpteen times. You can `ignore read-only`.

**Reply by ghost** [reply]
@Froggy232 The recent gvfs 1.48.0 version seems to have issues with admin:// according to [this bug report](https://bugs.archlinux.org/task/70108?project=1&string=gvfs). More details on the patch are [here](https://github.com/archlinux/svntogit-packages/commit/0200e7f592d1448333c407a32a41ecd121f19877#diff-3e341d2d9c67be01819b25b25d5e53ea3cdf3a38d28846cda85a195eb9b7203a). Don't know if that affects you, but it might give you something to go on for further investaigations.

**Reply by Froggy232** [reply]
@rusty-snake Thanks you a lot, it's perfect.
@glitsj16 Thanks, I will stick with `sudo gedit` for now

Thanks for your help, it's working pretty much perfectly now, and thanks you also for firejail, it's a wonderful software.
Bye!

**Reply by Froggy232** [reply]
Sorry, I still have the problem!
It work with a lot of file now, but some of them still doesn't work : user.js from mozilla profile, as example.

For information, I have `ignore read-only` and `ignore include disable-common.inc` in my gedit.local file.
It work if I launch with `firejail --noprofile gedit $FILE` or if I launch it with `sudo gedit $FILE`.

It's not a big deal, but if someone have an idea I would be happy to test it.
Thanks you

**Reply by rusty-snake** [reply]
For `~/.mozilla` (and the most other program dotfiles) it's `ignore include disable-programs.inc` resp. `noblacklist ${HOME}/.mozilla`.

**Reply by Froggy232** [reply]
It works perfectly, thanks you a lot
